### ISSUE TITLE
 [Lang Enhancement Proposal] Implement independent Para-C Pre-Processor and Grammar File

### ISSUE THREAD

**Issue by Luna-Klatzer** [Para-C/Para-C#5]
**Affected area by change:**
- Language Syntax: yes
- Language Library (Parac Base Library); no
- Pre-Processor: yes
- Documentation: yes

**Summary:**
Implement the Para-C Pre-Processor and separate the Antlr4 Grammar file, which currently implements both the preprocessor statements and standard language. This causes an issue when trying to parse conditional blocks, where statements are separated by the pre-processor statements:

_Example: (Currently is not allowed in the Para-C, due to the failed recognition of the pre-processor directives_
```c
if ('x' == 'x')
#ifdef X // Fails to recognise this directive due to the expected { }
{
    printf("Hello World");
}
#else
{
    printf("Not Hello World");   
}
#endif 
```

To achieve that the lexer and parser can recognise this, the processes need to be split, where an independent pre-processor lexer and parser are generated. These will then generate a new altered version of the code and pass it onto the actual Para-C Core Language Lexer and Parser.

**Required Changes:**
- Implement a basic Pre-Processor that alters the code and passes it to the Core Lexer and Parser
- Create Pre-Processor Grammar File, which ignores actual Para-C statements and syntax
- Generate Pre-Processor Lexer and Parser based on the Grammar and implement it into the structure

**Comment by Luna-Klatzer** [comment]
Implemented basic grammar structure with 19f5350c78d7817b37175539155e67e6be707581

**Comment by Luna-Klatzer** [comment]
Structure and Base Logic done with c2cbd02952da2a5e830656073eddf37fc6fb8f16. Only things left are implementing listener functions and code generation

**Comment by Luna-Klatzer** [comment]
The base issue is solved and already merged into the `main` branch since `v0.1.dev2`, therefore it will be closed and the code generation will be moved to a new issue
